sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/timeExpressions.scala

```diff
@@ -1029,7 +1085,14 @@ case class TimeToMillis(child: Expression)
 case class TimeToMicros(child: Expression)
   extends TimeToLongBase {
 
-  override def scaleFactor: Long = NANOS_PER_MICROS
+  override def nullSafeEval(input: Any): Any = {
+    DateTimeUtils.timeToMicros(input.asInstanceOf[Number].longValue())
+  }
+
+  override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
```

Comment:
it's not unified, can you follow `MakeTimestampNTZ` exactly?

Reply:
Yes, done